apps/page_mapping/src/main.c

```diff
@@ -77,8 +78,23 @@ static void inline prepare_page_table(seL4_Word addr, int npage, seL4_CPtr untyp
                                           seL4_ARCH_Default_VMAttributes);
         }
 
-#endif /* CONFIG_ARCH_X86_64 */
         assert(err == 0);
+#elif defined(CONFIG_ARCH_AARCH64)
+        while (err != seL4_DeleteFirst) {
+            pt = *free_slot;
+            err = untyped_retype_root(untyped, seL4_ARCH_PageTableObject,
+                                      seL4_PageTableBits, pt);
+            assert(err == 0);
+            (*free_slot)++;
+
+            err = seL4_ARCH_PageTable_Map(pt, SEL4UTILS_PD_SLOT, addr,
```

Comment:
Should'n there be an `assert(err == 0);` afterwards then also, just to play safe?